core/ai_orchestrator.go

```diff
@@ -397,18 +397,43 @@ type AIJobRequestData struct {
 }
 
 // CheckAICapacity verifies if the orchestrator can process a request for a specific pipeline and modelID.
-func (orch *orchestrator) CheckAICapacity(pipeline, modelID string) bool {
+func (orch *orchestrator) CheckAICapacity(pipeline, modelID string) (bool, chan<- bool) {
+	var hasCapacity bool
 	if orch.node.AIWorker != nil {
 		// confirm local worker has capacity
-		return orch.node.AIWorker.HasCapacity(pipeline, modelID)
+		err := orch.node.ReserveAICapability(pipeline, modelID)
+		if err == nil {
+			hasCapacity = true
+		}
 	} else {
 		// remote workers: RemoteAIWorkerManager only selects remote workers if they have capacity for the pipeline/model
 		if orch.node.AIWorkerManager != nil {
-			return orch.node.AIWorkerManager.workerHasCapacity(pipeline, modelID)
-		} else {
-			return false
+			hasCapacity = orch.node.AIWorkerManager.workerHasCapacity(pipeline, modelID)
 		}
 	}
+
+	if !hasCapacity {
+		return false, nil
+	} else {
+		releaseCapacity := make(chan bool)
+		// TODO: add capacity tracking to live-video-to-video pipeline with
+		if pipeline == "live-video-to-video" {
+			orch.node.ReleaseAICapability(pipeline, modelID)
+			close(releaseCapacity)
+			return true, nil
+		}
```

Comment:
I think we can just remove this because we already track capacity with this [containers](https://github.com/livepeer/go-livepeer/blob/cf8dbf503c2860e9f05f5d193df17b05aad95a4f/ai/worker/docker.go#L112) map. @victorges could you check?
We don't have any call to [ReserveAICapability](https://github.com/livepeer/go-livepeer/blob/cf8dbf503c2860e9f05f5d193df17b05aad95a4f/core/ai.go#L197) for live video so this call to `ReleaseAICapability` would cause the Capacity value to always increase and never decrease.

Reply:
I reworked the updates in this function to keep same functionality for live-video-to-video using local ai-worker.  This allowed removing this short circuit code noted above.